bpf/bpf_overlay.c

```diff
@@ -74,6 +74,21 @@ static __always_inline int handle_ipv6(struct __ctx_buff *ctx,
 		 */
 		if (*identity == HOST_ID)
 			return DROP_INVALID_IDENTITY;
+
+		/* Maybe overwrite the REMOTE_NODE_ID with
+		 * KUBE_APISERVER_NODE_ID to support upgrade. After v1.12,
+		 * this should be removed. */
+		if (*identity == REMOTE_NODE_ID) {
+			struct remote_endpoint_info *info;
+
+			/* Look up the ipcache for the src IP, it will give us
+			 * the real identity of that IP */
+			info = ipcache_lookup6(&IPCACHE_MAP,
+					       (union v6addr *) &ip6->saddr,
+					       V6_CACHE_KEY_LEN);
+			if (info)
+				*identity = info->sec_label;
+		}
```

Comment:
It would be good to get a look from the @cilium/sig-encryption team on this change.